frontend/src/metabase/query_builder/components/NativeQueryEditor/CodeMirrorEditor/CodeMirrorEditor.tsx

```diff
@@ -74,21 +78,17 @@ export const CodeMirrorEditor = forwardRef<
     (update: ViewUpdate) => {
       // handle selection changes
       const value = update.state.doc.toString();
+
       if (onSelectionChange) {
-        const beforeRange = convertSelectionToRange(
-          update.startState.doc.toString(),
-          update.startState.selection.main,
-        );
-        const afterRange = convertSelectionToRange(
-          value,
-          update.state.selection.main,
-        );
+        const beforeSelection = update.startState.selection.main;
```

Comment:
Looks like another merge conflict

Reply:
fixed with 0f1a2cc366e2c497c8e8f2759ad831928868b1fd

Reply:
there's a PR specifically for `metabot-v3-main` https://github.com/metabase/metabase/pull/53599, but after using the latest master code, this doesn't seem to happen anymore, so I didn't leave any of the code from #53599 in this branch.